src/onnx/parse_convolution.cpp

```diff
@@ -42,18 +43,201 @@ struct parse_convolution : op_parser<parse_convolution>
         return {{"Conv", "convolution"}, {"ConvInteger", "quant_convolution"}};
     }
 
+    // Convert to half prior to a shift to ensure we preserve accuracy here then
+    // convert back to int8
+    static instruction_ref add_int8_shift(const onnx_parser::node_info& info,
```

Comment:
Suggestion: a name like uint8_to_int8 might be more functionally informative. But this is highly subjective, so you don't have to change anything really.. up to you. Thanks.

Reply:
The intent here is to shift everything to int8 as most of our backend only works smoothly with int8 and not uint8, primarily MLIR with their kernels

Reply:
Yes, agreed on that: to convert to int8. But exactly what is to be (permissibly) converted? Only uint8. I assume this is to convert only from uint8 to int8. But never mind..